src/libs/ck-libs/io/ckio.C

```diff
@@ -1,436 +1,995 @@
+    if (!grp_ptr)
+    {
+      CkPrintf("The pointer to the local branch is null on pe=%d\n", CkMyPe());
+      CkExit();
+    }
+    grp_ptr->serveRead(bytes, offset, data, after_read, read_stride, num_readers);
+  }
 
-        void write(Session session, const char *data, size_t bytes, size_t offset) {
-          Options &opts = files[session.file].opts;
-          size_t stripe = opts.peStripe;
+  void write(Session session, const char* data, size_t bytes, size_t offset)
+  {
+    Options& opts = files[session.file].opts;
+    size_t stripe = opts.peStripe;
 
-          CkAssert(offset >= session.offset);
-          CkAssert(offset + bytes <= session.offset + session.bytes);
+    CkAssert(offset >= session.offset);
+    CkAssert(offset + bytes <= session.offset + session.bytes);
 
-          size_t sessionStripeBase = (session.offset / stripe) * stripe;
+    size_t sessionStripeBase = (session.offset / stripe) * stripe;
 
-          while (bytes > 0) {
-            size_t stripeIndex = (offset - sessionStripeBase) / stripe;
-            size_t bytesToSend = min(bytes, stripe - offset % stripe);
+    while (bytes > 0)
+    {
+      size_t stripeIndex = (offset - sessionStripeBase) / stripe;
+      size_t bytesToSend = min(bytes, stripe - offset % stripe);
 
-            CProxy_WriteSession(session.sessionID)[stripeIndex]
-              .forwardData(data, bytesToSend, offset);
+      CProxy_WriteSession(session.sessionID)[stripeIndex].forwardData(data, bytesToSend,
+                                                                      offset);
 
-            data += bytesToSend;
-            offset += bytesToSend;
-            bytes -= bytesToSend;
-          }
-        }
+      data += bytesToSend;
+      offset += bytesToSend;
+      bytes -= bytesToSend;
+    }
+  }
 
-        void doClose(FileToken token, CkCallback closed) {
-          int fd = files[token].fd;
-          if (fd != -1) {
-            int ret;
-            do {
+  void doClose(FileToken token, CkCallback closed)
+  {
+    int fd = files[token].fd;
+    if (fd != -1)
+    {
+      int ret;
+      do
+      {
 #if defined(_WIN32)
-              ret = _close(fd);
+        ret = _close(fd);
 #else
-              ret = ::close(fd);
+        ret = ::close(fd);
 #endif
-            } while (ret < 0 && errno == EINTR);
-            if (ret < 0)
-              fatalError("close failed", files[token].name);
-          }
-          files.erase(token);
-          contribute(closed);
-        }
+      } while (ret < 0 && errno == EINTR);
+      if (ret < 0)
+        fatalError("close failed", files[token].name);
+    }
+    files.erase(token);
+    contribute(closed);
+  }
 
-        int procNum(int arrayHdl,const CkArrayIndex &element)
-        {
+  int procNum(int arrayHdl, const CkArrayIndex& element)
+  {
 #if 0
           int peIndex = stripeIndex % opts.activePEs;
           int pe = opts.basePE + peIndex * opts.skipPEs;
 #endif
-          return 0;
-        }
+    return 0;
```

Comment:
Not clear from context why we're mapping all elements to 0.

Reply:
this is code from the output part written years ago - the git diff is being weird